community/boilerplates/auth-webhooks/lambda-cognito/tokenDecoder.js

```diff
@@ -0,0 +1,98 @@
+/*
+
+This code is taken from an AWS sample code repository. The code can be found here:
+
+https://github.com/awslabs/aws-support-tools/blob/master/Cognito/decode-verify-jwt/decode-verify-jwt.js
+
+It's seperated out into is own file and function for two reason.
+
+First, It's easier to read and possibly test.
+
+Second, Ther is a network request here to go download the secret keys from Cognito. According to lambda best practices here:
+
+https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html
+
+It's good practice to seperate out logic from the app entry point. This can make the function run faster especially when calls happen fast which I can imagine will be the case with this web hook.
+
+*/
+
+//Required to complete the AWS Code example
+var jose = require('node-jose');
+
+//In the lambda console you can define environment variables. https://docs.aws.amazon.com/lambda/latest/dg/env_variables.html
+const region = process.env.AWS_COGNITO_REGION;
+const cognitoPool = process.env.AWS_COGNITO_USER_POOL;
+
+/*
+Here we are using the URL that the AWS Documentation says you can get your JSON Web Key Set (JWKS)
+https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-verifying-a-jwt.html
+*/
+const keys_url = `https://cognito-idp.${region}.amazonaws.com/${cognitoPool}/.well-known/jwks.json`;
+const asyncRequest = require('httpsAsync.js');
+
+
+async function decode(token) {
+    /*
+    This is all from the sample AWS Code layed out in the first link above. The only difference is that I made an async/ await compatable get request for the token. I also provide 
+    but commented out a modified sample to compare the audience key in the token to multiple application client secrets. Application client secrets are made from the cognito console. 
+    The thoght is that there could be multiple client, each with their own keys, using Hasura.
+    */
+
+    //1) Parse out and decode the Key ID (kid) from the incomming token.
+    const sections = token.split(".");
+    var header = jose.util.base64url.decode(sections[0]);
+    header = JSON.parse(header);
```

Comment:
@Nomad-Go I'm not sure if you have to manually parse the parts of the token? Doesn't the `jose` library have APIs which let's you get the `kid` directly ?

Reply:
Let me change this code to see if I can get the KID using the `jose` library built in functions and test it in my companies development environment. I looks like AWS uses the `RS256` algorithm as described [here](https://aws.amazon.com/premiumsupport/knowledge-center/decode-verify-cognito-json-token/).

Reply:
I left another comment with a change to parse the kid using the `jose` library.